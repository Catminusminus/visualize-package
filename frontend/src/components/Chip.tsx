/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import { withStyles, WithStyles, Theme } from '@material-ui/core/styles'
import Chip from '@material-ui/core/Chip'
import { PropTypes, createStyles } from '@material-ui/core'

const styles = (theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      justifyContent: 'center',
      flexWrap: 'wrap'
    },
    chip: {
      margin: theme.spacing(1)
    }
  })

interface Props extends WithStyles<typeof styles> {
  label: React.ReactNode
  onDelete: (event: any) => void
  color: PropTypes.Color
}

const ChipComponent: React.FC<Props> = ({
  classes,
  label,
  onDelete,
  color
}) => (
  <div className={classes.root}>
    <Chip
      label={label}
      onDelete={onDelete}
      className={classes.chip}
      color={color}
    />
  </div>
)

export default withStyles(styles)(ChipComponent)
