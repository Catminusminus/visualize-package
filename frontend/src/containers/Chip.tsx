/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { State, deletePackage } from '../modules'
import Paper from '@material-ui/core/Paper'
import { withStyles, createStyles, Theme } from '@material-ui/core/styles'
import Chip from '@material-ui/core/Chip'

const styles = createStyles((theme: Theme) => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
    padding: theme.spacing(1 / 2)
  },
  chip: {
    margin: theme.spacing(1 / 2)
  }
}))

const packagesSelector = (state: State) => state.reducer.packages

const ChipContainer = ({ classes }) => {
  const packages = useSelector(packagesSelector)
  const dispatch = useDispatch()
  return (
    <Paper className={classes.root}>
      {packages.map(packageName => (
        <Chip
          key={packageName}
          className={classes.chip}
          color={'primary'}
          label={packageName}
          onDelete={() => dispatch(deletePackage({ packageName }))}
        />
      ))}
    </Paper>
  )
}

export default withStyles(styles)(ChipContainer)
