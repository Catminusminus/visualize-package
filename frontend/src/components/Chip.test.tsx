/**
 * Author of this code work, Yuki Mogi. c FFRI, Inc. 2019
 */

import * as React from 'react'
import renderer from 'react-test-renderer'
import Chip from './Chip'

it('Snapshot test of Chip', () => {
  const result = renderer
    .create(<Chip label="@ffri/sample" onDelete={() => {}} color="primary" />)
    .toJSON()
  expect(result).toMatchSnapshot()
})
