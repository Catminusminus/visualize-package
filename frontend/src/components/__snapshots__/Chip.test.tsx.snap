// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of Chip 1`] = `
<div
  className="ChipComponent-root-1"
>
  <div
    className="MuiChip-root ChipComponent-chip-2 MuiChip-colorPrimary MuiChip-deletableColorPrimary MuiChip-deletable"
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label"
    >
      @ffri/sample
    </span>
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</div>
`;
